### PR TITLE
Add composite RPi driver for pcm512x DAC HAT cards

### PR DESCRIPTION
**Pull request by timg236**
It might be possible to add other drivers but I started with these smaller drivers first.
* Allo Piano DAC
* DionAudio Loco V2
* Justboom DAC
* IQAudioIO DAC

I've tested this with the Justboom DAC so it would be great if anyone could test the other cards.

@HiassofT @iqaudio @allocom

### CODE REVIEW

**Comment by iqaudio** [comment]
Prefer not to be merged as I’d like to ensure anything I change for IQaudIO doesn’t effect others and vice versa. Such a change doesn’t feel right.

Gordon

**Comment by pelwell** [comment]
Don't worry Gordon, I promise we'll help you with that when the day comes.

**Comment by iqaudio** [comment]
Well if it’s okay with you Phil, it’s okay with me! :-)

I’m a little swamped at the moment so unable to test. Sorry.

Gordon

**Comment by pelwell** [comment]
@timg236 is one of us, and he knows what he's doing.

**Comment by HiassofT** [comment]
IQaudio DAC worked fine here, code looks good to me as well, thanks a lot!

**Comment by pelwell** [comment]
If you rebase against the latest 4.18 commits I'll merge it.

File: sound/soc/bcm/rpi-pcm512x-soundcard.c [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+{
+	.name        = "Piano DAC",
+	.stream_name = "Piano DAC HiFi",
+},
+};
+
+static struct snd_rpi_pcm512x_drvdata drvdata_allo_piano_dac = {
+	.card_name                = "PianoDAC",
+	.dai                      = snd_allo_piano_dac_dai,
+	.digital_gain_0db_name_dt = "alloc,24db_digital_gain",
```
Comment (timg236) [inline sound/soc/bcm/rpi-pcm512x-soundcard.c]: typo: "allo" not "alloc"

**Comment by timg236** [comment]
> If you rebase against the latest 4.18 commits I'll merge it.

done

**Comment by pelwell** [comment]
Merged - thanks, everyone.

**Comment by timg236** [comment]
> Merged - thanks, everyone.

Thanks!

**Comment by allocom** [comment]
Hi timg236 & pelwell,
Had some problem with forwarding of emails, just observed on github and tested with our piano card. I see "compatible" string mismatch, it should be "allo,piano-dac" instead of "allo,allo-piano-dac" at line no 257. 
What do you suggest to do? You can make the corrections OR want us to send a pull request ??

**Comment by timg236** [comment]
Ah, sorry. I'll fix that now

**Comment by allocom** [comment]
Thanks timg236

**Pull request merged**